app/code/Magento/Swatches/Block/Product/Renderer/Configurable.php

```diff
@@ -200,7 +210,20 @@ public function getJsonSwatchConfig()
     public function getNumberSwatchesPerProduct()
     {
         return $this->_scopeConfig->getValue(
-            'catalog/frontend/swatches_per_product',
+            self::XML_PATH_SWATCHES_PER_PRODUCT,
+            ScopeInterface::SCOPE_STORE
+        );
+    }
+
+    /**
+     * Get config if swatch tooltips should be rendered.
+     *
+     * @return string
+     */
+    public function getShowSwatchTooltip()
```

Comment:
Unfortunately, we are not allowed to add new public methods according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
@sidolov Where do you read that? Could you please quote that?

Reply:
@torhoehn I'm not sure that such case is documented but introducing the new public method in api class may break 3d party implementation that extended the current class and defined the method with the same name but with a different signature. We do not recommend inheritance that may cause such error but 3d party developers still use this pattern and we should guarantee compatibility with our code base.

Reply:
Wait. So we cannot add new methods, a rule that is not documented/written anywhere, because some hypothetical third-party extension that, against all official advice that has been written down, extends this exact class _might_ under some unforeseen circumstances produce errors?
If - and only if - we accept this, what would be the alternative way to go about this, then? How can this be the criterion, in turn hindering further development?

Reply:
Are we still open source or do we have always have to "ask" internal Magento/Adobe people if the specific code change is acceptable? 

It's not Magento's responsibility to support ugly written code from 3rd parties. The only responsibility is to have documented changes, everybody can look at.

Reply:
@sidolov @josefbehr Backward compatible guid don't forbid to add new public methods

> Adding parameters in public methods
> **Deprecate the method and add a new method with the new parameter(s) instead of adding them to a public method.**
> **Follow the alternative implementation described earlier for introducing a new method to a class or interface.**
> Reference the new method in a @see tag as a recommended replacement. Explain the reasons for replacing the old method with the new one (e.g., there is a bug in the old method).

Reply:
@josefbehr , @Den4ik , @torhoehn please, refer to the [PHP section of backward compatible development guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php): 

> **Introduction of a method to a class or interface**
> Create a new interface with a new method instead of introducing a method to a class or interface.

Reply:
@sidolov Thanks for we quote, haven't seen it. I updated the PR and added a view model, I hope this is fine for this case.

Reply:
@torhoehn thanks! You find a really good solution om this case. It would be great to cover the new functionality with automated tests